lib/shared/common.ts

```diff
@@ -925,29 +927,31 @@ export function createChannel(streamIn: StreamIn): StreamOut {
               if (result != null) {
                 if (typeof result !== 'object') throw new Error(`Expected onLoad() callback in plugin ${JSON.stringify(name)} to return an object`);
                 let keys: OptionKeys = {};
-                let pluginName = getFlag(result, keys, 'pluginName', mustBeString);
                 let contents = getFlag(result, keys, 'contents', mustBeStringOrUint8Array);
-                let resolveDir = getFlag(result, keys, 'resolveDir', mustBeString);
-                let pluginData = getFlag(result, keys, 'pluginData', canBeAnything);
-                let loader = getFlag(result, keys, 'loader', mustBeString);
-                let errors = getFlag(result, keys, 'errors', mustBeArray);
-                let warnings = getFlag(result, keys, 'warnings', mustBeArray);
-                let watchFiles = getFlag(result, keys, 'watchFiles', mustBeArray);
-                let watchDirs = getFlag(result, keys, 'watchDirs', mustBeArray);
-                checkForInvalidFlags(result, keys, `from onLoad() callback in plugin ${JSON.stringify(name)}`);
-
-                response.id = id;
-                if (pluginName != null) response.pluginName = pluginName;
-                if (contents instanceof Uint8Array) response.contents = contents;
-                else if (contents != null) response.contents = protocol.encodeUTF8(contents);
-                if (resolveDir != null) response.resolveDir = resolveDir;
-                if (pluginData != null) response.pluginData = stash.store(pluginData);
-                if (loader != null) response.loader = loader;
-                if (errors != null) response.errors = sanitizeMessages(errors, 'errors', stash, name);
-                if (warnings != null) response.warnings = sanitizeMessages(warnings, 'warnings', stash, name);
-                if (watchFiles != null) response.watchFiles = sanitizeStringArray(watchFiles, 'watchFiles');
-                if (watchDirs != null) response.watchDirs = sanitizeStringArray(watchDirs, 'watchDirs');
-                break;
+                if (contents != null) {
```

Comment:
This if statement can be inverted to reduce nesting.  This will make the change a lot more readable.